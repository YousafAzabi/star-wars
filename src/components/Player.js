import React from 'react';

import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import { withStyles } from '@material-ui/core/styles';

import 'bootstrap/dist/css/bootstrap.css';

import config from '../config';

const styles = theme => ({
  card: {
    margin: '5%',
    float: 'left',
    width: '40%',
    textAlign: 'left',
    border: 'groove 1px',
    borderRadius: 10
  },
  badge: {
    float: 'right',
    right: '80px'
  },
});

class Player extends React.Component {

  render() {
    const { classes, isDisplayed, title, type, details, winningAttribute } = this.props;
    const attributes = config.ATTRIBUTES[type];

    return (
      <div className={classes.card}>
        <div className="card-body">
          <h5 className="card-title" style={{textAlign: 'center'}}>
            {title}
            <span className="badge badge-secondary">{winningAttribute}</span>
          </h5>
          {details !== null &&
            <List className={classes.details}>
            { attributes.map(attribute =>
              <ListItem key={attribute}>
                <b>{attribute}</b>: {isDisplayed && details[attribute]}
              </ListItem>
            )}
            </List>
          }
        </div>
      </div>
    );
  }
}

export default withStyles(styles)(Player);
