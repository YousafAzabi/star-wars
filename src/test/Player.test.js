import React from "react";
import { create } from "react-test-renderer";
import Player from "../components/Player";

describe("Player component", () => {
  test("it matches the snapshot", () => {
    const component = create(<Player
      title='Player One'
      type='people'
      details={{name: 'foo', mass: 20, height: 50}}
      winningAttribute='mass'/>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("test title text", () => {
    const component = create(<Player
      title='Player One'
      type='people'
      details={{name: 'foo', mass: 20, height: 50}}
      winningAttribute='mass'/>
    );
    const rootInstance = component.root;
    const title = rootInstance.findByType('h5');
    expect(title.props.children[0]).toBe('Player One');
  });

  test("test badge text", () => {
    const component = create(<Player
      title='Player One'
      type='people'
      details={{name: 'foo', mass: 20, height: 50}}
      winningAttribute='mass'/>
    );
    const rootInstance = component.root;
    const badge = rootInstance.findByType('span');
    expect(badge.props.children).toBe('mass');
  });

});
