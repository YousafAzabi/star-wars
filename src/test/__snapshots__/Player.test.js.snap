// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component it matches the snapshot 1`] = `
<div
  className="Player-card-1"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Player One
      <span
        className="badge badge-secondary"
      >
        mass
      </span>
    </h5>
    <ul
      className="MuiList-root MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <b>
          height
        </b>
        : 
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <b>
          mass
        </b>
        : 
      </li>
    </ul>
  </div>
</div>
`;
